metrics/isolator.hpp

```diff
@@ -0,0 +1,180 @@
+#ifndef __METRICS_ISOLATOR_MODULE_HPP__
+#define __METRICS_ISOLATOR_MODULE_HPP__
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <mesos/mesos.hpp>
+#include <mesos/module.hpp>
+
+#include <mesos/slave/isolator.hpp>
+
+#include <process/future.hpp>
+#include <process/owned.hpp>
+#include <process/process.hpp>
+
+#include <stout/duration.hpp>
+#include <stout/flags.hpp>
+#include <stout/hashset.hpp>
+#include <stout/json.hpp>
+#include <stout/option.hpp>
+#include <stout/try.hpp>
+
+#include <stout/flags/parse.hpp>
+
+namespace mesosphere {
+namespace dcos {
+namespace metrics {
+
+template<typename T>
+Try<T> parse(const std::string& body)
+{
+  Try<JSON::Object> json = flags::parse<JSON::Object>(body);
+  if (json.isError()) {
+    return Error("Error parsing input as JSON: " + json.error());
+  }
+
+  Try<T> response = ::protobuf::parse<T>(json.get());
+  if (response.isError()) {
+    return Error(
+        "Error parsing JSON into protobuf: " + response.error());
+  }
+
+  return response.get();
+}
+
+// Forward declaration.
+class MetricsIsolatorProcess;
+
+namespace isolator {
+
+struct Flags : virtual flags::FlagsBase
+{
+  Flags()
+  {
+    add(&Flags::service_scheme,
```

Comment:
I think I'd prefer to leave this in place for now, since I use it in the tests. When I have some time I could update the tests with a mock metrics service that listens on a domain socket. WDYT?